### PR TITLE
[FIX JENKINS-41457] Use BUILD_NOW_TEXT for parameterised jobs

### PR DESCRIPTION
**Pull request by alvarolobato**
https://issues.jenkins-ci.org/browse/JENKINS-41757

@reviewbybees @jenkinsci/code-reviewers

### CODE REVIEW

**Comment by recampbell** [comment]
Asking @jglick to take a look

File: test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java [hunk 1]
```diff
@@ -58,9 +60,27 @@
...
+     */
+    @Test
+    public void basicsWithParameter() throws Exception {
+        Impl.oldschool = false;
+        FreeStyleProject p = j.createFreeStyleProject("aaa");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("FOO", null)));
+        String pageText = j.createWebClient().getPage(p).asText();
+        assertThat(pageText, containsString("newschool:aaa"));
+
+        Impl.oldschool = true;
```
Comment (jglick) [inline test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java]: use a `try`-`finally` block

File: test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java [hunk 2]
```diff
@@ -58,9 +60,27 @@
...
         assertThat(j.createWebClient().getPage(p).asText(), containsString("newschool:aaa"));
+
         Impl.oldschool = true;
         assertThat(j.createWebClient().getPage(p).asText(), containsString("oldschool:aaa"));
     }
+
+    /**
+     * Makes sure that {@link AlternativeUiTextProvider} actually works with a parameterized Job.
+     */
+    @Test
```
Comment (jglick) [inline test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java]:
```java
@Issue("JENKINS-41757")
```

**Review by jglick — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
LGTM 👍

File: test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java (hunk 2 above)
Comment (alvarolobato) [inline test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java]: Yeah... forgot it.

File: test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java (hunk 1 above)
Comment (alvarolobato) [inline test/src/test/java/hudson/util/AlternativeUiTextProviderTest.java]: @jglick sorry but I don't know what the try-finally is for

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java [hunk 3]
```diff
@@ -278,7 +278,8 @@ public static final CauseAction getBuildCause(ParameterizedJob job, StaplerReque
      * Uses {@link #BUILD_NOW_TEXT}.
      */
     public final String getBuildNowText() {
-        return isParameterized() ? Messages.ParameterizedJobMixIn_build_with_parameters() : AlternativeUiTextProvider.get(BUILD_NOW_TEXT, asJob(), Messages.ParameterizedJobMixIn_build_now());
+        return isParameterized() ? AlternativeUiTextProvider.get(BUILD_NOW_TEXT, asJob(), Messages.ParameterizedJobMixIn_build_with_parameters())
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Shouldn't this rather have a new variant `BUILD_WITH_PARAMETERS_TEXT`? The difference IIUC is that "Build Now" indicates an immediate action, while the other does not (like when in many applications some actions have a trailing `…` and others don't).

Comment (alvarolobato) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Not needed, `BUILD_NOW_TEXT` is the `AlternativeUiTextProvider` inside it you can check if the job has parameters and decide what text you want to display.

Comment (daniel-beck) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Have a thumbs up then: :+1:

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Review by jglick — APPROVED** [review]

**Pull request merged**